exonum-java-binding-cryptocurrency-demo/src/main/java/com/exonum/binding/cryptocurrency/CryptocurrencyServiceImpl.java

```diff
@@ -97,6 +106,23 @@ public HashCode submitTransaction(Transaction tx) {
     });
   }
 
+  @Override
+  public List<TransferTxData> getWalletHistory(PublicKey ownerKey) {
+    checkBlockchainInitialized();
+
+    return node.withSnapshot(view -> {
+      CryptocurrencySchema schema = new CryptocurrencySchema(view);
+      ProofListIndexProxy<TransferTxData> history = schema.walletHistory(ownerKey);
+
+      List<TransferTxData> result = new ArrayList<>();
+      for (TransferTxData tx : history) {
```

Comment:
Streams instead of foreach ?

Reply:
`ProofListIndexProxy` doesn't support streams api.